esx_service/utils/vmdk_utils.py

```diff
@@ -163,11 +163,14 @@ def get_volumes(tenant_re):
                     logging.debug("get_volumes: cannot find tenant_name for tenant_uuid=%s", sub_dir_name)
                     logging.debug("get_volumes: path=%s root=%s sub_dir_name=%s",
                                 path, root, sub_dir_name)
-                    for file_name in list_vmdks(root):
-                        volumes.append({'path': root,
-                                        'filename': file_name,
-                                        'datastore': datastore,
-                                        'tenant' : auth_data_const.ORPHAN_TENANT})
+
+                    # return orphan volumes only in case when volumes from any tenants are asked
+                    if tenant_re == "*":
+                        for file_name in list_vmdks(root):
+                            volumes.append({'path': root,
+                                            'filename': file_name,
+                                            'datastore': datastore,
```

Comment:
btw , L168-173 more python-ish would be something like this  
```
volumes += [ {'path': root, 'filename': file_name, 'datastore': datastore,  
                    'tenant' : auth_data_const.ORPHAN_TENANT} for  file_name in list_vmdks(root)]
```

also applies to other places in this function